#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_2.6.11.dpatch by  <rstuart@ras.pc.brisbane.lube>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make work on kernels 2.6.11 and greater

@DPATCH@
--- btsco-0.42.orig/kernel/btsco.c
+++ btsco-0.42/kernel/btsco.c
@@ -94,7 +94,12 @@
 
 static int index[SNDRV_CARDS] = {[0 ... (SNDRV_CARDS - 1)] = -2}; /* Exclude the first card */
 
+#if	LINUX_VERSION_CODE < KERNEL_VERSION(2,6,11)
+static int card_count;
+module_param_array(index, int, card_count, 0444);
+#else
 module_param_array(index, int, NULL, 0444);
+#endif
 MODULE_PARM_DESC(index, "Index value for Bluetooth SCO Headset Soundcard.");
 
 #undef dprintk
